plutus-core/plutus-ir/test/PlutusIR/Transform/StrictLetRec/strictLetRec.golden

```diff
@@ -0,0 +1,329 @@
+(let
```

Comment:
Can we print out UPLC, instead of TPLC? The TPLC is 100x harder to read, and for no benefit.

Reply:
It's not just harder to read, it's also noisy. It has this huge `fixBy` definition that is generated "just in case", which is normally removed via dead code elimination, but not in this case because we turn optimizations off.

(Note that some work in this PR is done by me and some by Yura, so it's not immediately obvious who reviewers are talking to)

Reply:
> Can we print out UPLC, instead of TPLC? The TPLC is 100x harder to read, and for no benefit.

The reason I wrote PIR in this case is because I was following this advice by @michaelpj 
![image](https://github.com/IntersectMBO/plutus/assets/1009751/ad8b3e19-06d8-4bc2-a6a7-bcf73ed93a5f)

This test isolates just one pass PIR -> PIR, so that we can reason how PIR compilation treats Let(rec)s without taking into the equation possible effects added by compilation to UPLC.

Reply:
Right, this is PIR, not TPLC.. but still, this is too much to look at. At the very least you should print _readable_ PIR, but it will still be too big. I think the UPLC for this program should be quite small, probably 10 lines or so.